core/src/test/java/org/apache/iceberg/util/TestSnapshotUtil.java

```diff
@@ -75,9 +75,9 @@ public void before() throws Exception {
 
     this.table = TestTables.create(tableDir, "test", SCHEMA, SPEC, 2);
     table.newFastAppend().appendFile(FILE_A).commit();
-    this.snapshotAId = table.currentSnapshot().snapshotId();
-
-    snapshotATimestamp = System.currentTimeMillis();
+    Snapshot snapshotA = table.currentSnapshot();
+    this.snapshotAId = snapshotA.snapshotId();
+    this.snapshotATimestamp = snapshotA.timestampMillis() + 1;
```

Comment:
I am afraid it does not guarantee the timestamp of snapshot B. Would it be more reliable to use `waitUntilAfter`?

```
Snapshot snapshotA = table.currentSnapshot();
this.snapshotAId = snapshotA.snapshotId();
this.snapshotATimestamp = snapshotA.timestampMillis();

TestHelpers.waitUntilAfter(snapshotATimestamp);

...
```

Then we can adapt the test to call `SnapshotUtil.oldestAncestorAfter(table, snapshotATimestamp + 1)`.

Reply:
Great. Done.